programs/perpetuals/src/state/staking.rs

```diff
@@ -0,0 +1,120 @@
+use {
+    super::{cortex::SECONDS_PER_HOURS, user_staking::UserStaking},
+    crate::math,
+    anchor_lang::prelude::*,
+};
+
+#[derive(Copy, Clone, PartialEq, AnchorSerialize, AnchorDeserialize, Debug)]
+pub enum StakingType {
+    LM = 1,
+    LP = 2,
+}
+
+impl Default for StakingType {
+    fn default() -> Self {
+        Self::LM
+    }
+}
+
+#[account]
+#[derive(Default, Debug)]
+pub struct Staking {
+    //
+    // Type of staking
+    //
+    pub staking_type: StakingType,
+    //
+    // Bumps
+    //
+    pub bump: u8,
+    pub staked_token_vault_bump: u8,
+    pub reward_token_vault_bump: u8,
+    pub lm_reward_token_vault_bump: u8,
+    //
+    // Tokens in locked stake
+    //
+    pub nb_locked_tokens: u128,
```

Comment:
Since the max amount of token a token account can hold is u64, this could be u64 too isn't it? (in native denomination) https://github.com/solana-labs/solana-program-library/blob/2ccc68ab14f041210930a5fbc64e14789fca3c6e/token/program-2022/src/state.rs#L98

Reply:
Same for other u128 values. I think where it's not accumulating over time for like accounting purpose, it should be u64?